components/micro-gateway-core/src/main/ballerina/src/gateway/filters/analytics_request_filter.bal

```diff
@@ -20,19 +20,23 @@ import ballerina/runtime;
 public type AnalyticsRequestFilter object {
 
     public function filterRequest(http:Caller caller, http:Request request, http:FilterContext context) returns boolean {
+        string amAnalyticsVersion = getAnalyticsVertion();
```

Comment:
Let's not call getAnalyticsVertion() method for each request . Save it in a global variable during startup and reuse it

Reply:
Fixed in https://github.com/wso2/product-microgateway/pull/1319/commits/b694c6960318d34763d343da195320faf623a55f